src/tool_cb_hdr.c

```diff
@@ -104,42 +104,17 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
     /* match only header that start with etag (case insensitive) */
     if(curl_strnequal(str, "etag:", 5)) {
       char *etag_h = NULL;
-      char *first = NULL;
-      char *last = NULL;
       size_t etag_length = 0;
+      etag_h = ptr + strlen("ETag:");
 
-      etag_h = ptr;
-      /* point to first occurrence of double quote */
-      first = memchr(etag_h, '\"', cb);
-
-      /*
-       * if server side messed with the etag header and doesn't include
-       * double quotes around the etag, kindly exit with a warning
-       */
-
-      if(!first) {
-        warnf(per->config->global,
-              "Received header etag is missing double quote/s\n");
-        return failure;
-      }
-      else {
-        /* discard first double quote */
-        first++;
-      }
-
-      /* point to last occurrence of double quote */
-      last = memchr(first, '\"', cb);
-
-      if(!last) {
-        warnf(per->config->global,
-              "Received header etag is missing double quote/s\n");
-        return failure;
-      }
+      /* pass all spaces */
+      while(*etag_h && ISSPACE(*etag_h))
+        etag_h++;
 
       /* get length of desired etag */
-      etag_length = (size_t)last - (size_t)first;
+      etag_length = (size_t)end - strlen("\r\n") - (size_t)etag_h;
```

Comment:
You must not assume CRLF!

Reply:
this is specified in HTTP that headers are separated with CRLF. I don't know why eol was added to the input line, maybe this should be fixed on the upper level. If you know how to fix that please do.

Reply:
curl needs to handle headers ending with CRLF or only LF

Reply:
not sure what to answer here: the input string have CRLF and we need to strip it. If you know how to fix it gracefully then push a commit to the branch

Reply:
The input string will have CRLF or just LF without any CR and the code needs to handle both cases.

Reply:
ok, now I see. Still not clear why header may have LF but if you are sure that this may happen then we should strip it gracefully. But maybe this should be fixed on top level i.e. the input `ptr` should be passed into the func with already trimmed EOL

Reply:
It is used in the curl test suite and very common "in the wild" too.

> maybe this should be fixed on top level i.e. the input ptr should be passed into the func with already trimmed EOL

Possibly, but that change should not be done within this PR =)

Reply:
ok, I'll add a commit with EOL skip